### PR TITLE
Unify TestStream and TestBatch SQL connectors

### PR DESCRIPTION
**Pull request by viliam-durina**
These connectors were almost same, extract the code into an abstract class.

### CODE REVIEW

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlImposeOrderFunctionTest.java [hunk 1]
```diff
@@ -295,4 +297,20 @@ private static String createTable(Object[]... values) {
...
+        return timestampTz(epochMillis).toLocalDate();
+    }
+
+    public static LocalDateTime timestamp(long epochMillis) {
+        return timestampTz(epochMillis).toLocalDateTime();
+    }
+
+    public static OffsetDateTime timestampTz(long epochMillis) {
+        return OffsetDateTime.ofInstant(Instant.ofEpochMilli(epochMillis), ZoneOffset.UTC);
+    }
```
Comment (gierlachg) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlImposeOrderFunctionTest.java:315]: Those don't need to be public. However, I envision them being used in other tests as well but in that case they should be moved.

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                    row[i] = null;
+                } else {
+                    row[i] = fields.get(i).getType().convert(values[i]);
+                }
+            }
+            rows.add(row);
+        }
+
+        boolean streaming = Boolean.parseBoolean(options.get(OPTION_STREAMING));
+        return new TestBatchTable(this, schemaName, mappingName, fields, rows, streaming);
```
Comment (gierlachg) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:195]: Why `TestBatchTable` is always used?

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    ) {
+        EventTimePolicy<Object[]> eventTimePolicy = eventTimePolicyProvider == null
+                ? EventTimePolicy.noEventTime()
+                : eventTimePolicyProvider.apply(null);
+
+        TestBatchTable table = (TestBatchTable) table_;
+        List<Object[]> rows = table.rows;
+        boolean streaming = table.streaming;
+
+        FunctionEx<Context, TestBatchDataGenerator> createContextFn = ctx -> {
```
Comment (gierlachg) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:215]: Why `TestBatchDataGenerator` is always used?

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+            ExpressionEvalContext evalContext = SimpleExpressionEvalContext.from(ctx);
+            return new TestBatchDataGenerator(rows, predicate, projection, evalContext, streaming);
+        };
+
+        ProcessorMetaSupplier pms;
+        if (table.streaming) {
```
Comment (gierlachg) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:221]: This class is abstract yet it contains logic belonging to its implementors... maybe get rid of `table.streaming` and move implementor specific logic to respective connectors?

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java (hunk 2 above)
Comment (viliam-durina) [reply hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:195]: Renamed it to `TestTable`

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java (hunk 3 above)
Comment (viliam-durina) [reply hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:215]: Renamed to `DataGenerator`

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java (hunk 4 above)
Comment (viliam-durina) [reply hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestAbstractSqlConnector.java:221]: I moved some code down, but not all. This way it's less code.

**Review by gierlachg — APPROVED** [review]

**Pull request merged**